x/oracle/client/common/proof/multi_result.go

```diff
@@ -1,215 +1,215 @@
-// 				OracleDataMultiProof: oracleDataList,
-// 				BlockRelayProof:      blockRelay,
-// 			},
-// 			EVMProofBytes: evmProofBytes,
-// 		})
-// 	}
-// }
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	clientcmn "github.com/bandprotocol/chain/x/oracle/client/common"
+	"github.com/bandprotocol/chain/x/oracle/types"
+	ics23 "github.com/confio/ics23/go"
+	"github.com/cosmos/cosmos-sdk/client"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+)
+
+type JsonMultiProof struct {
+	BlockHeight          uint64            `json:"block_height"`
+	OracleDataMultiProof []OracleDataProof `json:"oracle_data_multi_proof"`
+	BlockRelayProof      BlockRelayProof   `json:"block_relay_proof"`
+}
+
+type MultiProof struct {
+	JsonProof     JsonMultiProof   `json:"json_proof"`
+	EVMProofBytes tmbytes.HexBytes `json:"evm_proof_bytes"`
+}
+
+func GetMutiProofHandlerFn(cliCtx client.Context, route string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		requestIDs := r.URL.Query()["id"]
+		ctx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, cliCtx, r)
+		if !ok {
+			return
+		}
+		height := &ctx.Height
+		if ctx.Height == 0 {
+			height = nil
+		}
+
+		commit, err := ctx.Client.Commit(context.Background(), height)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+		signatures, err := GetSignaturesAndPrefix(&commit.SignedHeader)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		blockRelay := BlockRelayProof{
+			BlockHeaderMerkleParts: GetBlockHeaderMerkleParts(commit.Header),
+			Signatures:             signatures,
+		}
+
+		oracleDataBytesList := make([][]byte, len(requestIDs))
+		oracleDataList := make([]OracleDataProof, len(requestIDs))
+
+		isFirstRequest := true
+		for idx, requestID := range requestIDs {
+			intRequestID, err := strconv.ParseUint(requestID, 10, 64)
+			if err != nil {
+				rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+				return
+			}
+			requestID := types.RequestID(intRequestID)
+			bz, _, err := ctx.Query(fmt.Sprintf("custom/%s/%s/%d", route, types.QueryRequests, requestID))
+			if err != nil {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+				return
+			}
+			var qResult types.QueryResult
+			if err := json.Unmarshal(bz, &qResult); err != nil {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+				return
+			}
+			if qResult.Status != http.StatusOK {
+				clientcmn.PostProcessQueryResponse(w, ctx, bz)
+				return
+			}
+			var request types.QueryRequestResult
+			if err := ctx.LegacyAmino.UnmarshalJSON(qResult.Result, &request); err != nil {
```

Comment:
Legacy?

Reply:
changed to json.Unmarshal and got request.Result as follow: `client_id:"from_bandd" calldata:"\000\000\000\000\000\000\000d" resolve_status:RESOLVE_STATUS_SUCCESS result:"\000\000\000\000\000\002\310\211" `